hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -284,9 +287,11 @@ private boolean checkContainersReplicatedOnNode(DatanodeDetails dn)
         if (replicaSet.isSufficientlyReplicated()) {
           sufficientlyReplicated++;
         } else {
+          underReplicatedIDs.add(cid);
```

Comment:
I think we should wrap this line in `if (LOG.isDebugEnabled()) {` too - as if debug is not enabled, we never use the lists, so there is no point in adding to them.

Also, I wonder if we should set an upper bound on the list? Eg if there are 200,000 under replicated containers, is it really useful to log them all? Maybe logging just the first 10 or 100 would be enough?

Reply:
I added a limit of 10000 as the threshold.  From my experience of cluster maintenance,  I only care about the left over under replicated and unhealthy containers, which are of size from dozens to hundreds, so 10000 limit seems enough.